### PR TITLE
Change db builder to read new data schema with parent if

### PR DESCRIPTION
**Pull request by Chandrasekar-Rajasekar**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: db-builder/src/main/java/com/yahoo/covid19/database/DatabaseBuilder.java [hunk 1]
```diff
@@ -70,6 +70,7 @@
     private static final String CREATE_PLACE = "CREATE TABLE IF NOT EXISTS place\n" +
             "(\n" +
             "    id VARCHAR(255) DEFAULT NOT NULL,\n" +
+            "    type VARCHAR(255) DEFAULT NULL,\n" +
```
Comment (aklish) [inline db-builder/src/main/java/com/yahoo/covid19/database/DatabaseBuilder.java:73]: We'll probably need to make this a list of types - check out JsonType in Elide code to see how to turn a string into a JSON map in the Elide models.

File: db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java [hunk 2]
```diff
@@ -53,23 +53,24 @@
     private final String longitude;
     private final String latitude;
     private final String population;
-    private final String stateId;
-    private final String countryId;
+    private final List<String> parentId;
```
Comment (aklish) [inline db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java]: Pluralize this

File: db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java [hunk 3]
```diff
@@ -53,23 +53,24 @@
...
 
     @JsonAdapter(TableNameAdapter.class)
     private final JoinTableNames type;
 
     private transient ErrorCodes errorCode = OK;
+    private transient Boolean isValid = null;
```
Comment (aklish) [inline db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java]: This see like a hack - why do we need state for this? We already have an isValid method.

File: db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java [hunk 4]
```diff
@@ -85,126 +86,76 @@ private PreparedStatement getRelationshipStatement(DatabaseBuilder.DBConnector c
...
+            errorCode = INVALID_ID;
             return false;
         }
-        return true;
-    }
 
-    public boolean validateState(Map<String, Insertable> foreignKeyMap) {
-        if (! isValidCoordinate(latitude) || ! isValidCoordinate(longitude)) {
-            errorCode = INVALID_COORDINATE;
+        if (type == null) {
```
Comment (aklish) [inline db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java]: I'd remove the isValidRegion function and just move all this logic back into isValid.

File: db-builder/src/test/java/com/yahoo/covid19/database/PlacesTest.java [hunk 5]
```diff
@@ -6,19 +6,26 @@
...
 public class PlacesTest {
 
     @ParameterizedTest
     @CsvSource({"-120.66384, 40.44611, true", "1.2E10, 40.44611, true",  "40.44611, 1.2E10, true", "40, 1.ab0, false"})
     public void testLongitudeParsing(String longitude, String latitude, Boolean expected) {
-        Places places = new Places("123456", "", "", longitude, latitude, "0", null, null, JoinTableNames.Country);
-        Assertions.assertEquals(expected, places.isValid(null));
+        Places earth = new Places("000000", "", "", null, null, "0", null, JoinTableNames.Supername);
+        Places places = new Places("123456", "", "", longitude, latitude, "0", Arrays.asList("000000"), JoinTableNames.Country);
+        Map<String, Insertable> foreignKeyMap = new HashMap<>();
```
Comment (aklish) [inline db-builder/src/test/java/com/yahoo/covid19/database/PlacesTest.java:26]: Let's not mixup test cases.  I would make a new tests for this.

File: webservice/src/main/java/com/yahoo/covid19/models/Place.java [hunk 6]
```diff
@@ -21,6 +22,8 @@
 public class Place {
     @Id
     private String id;
+    @Column(name = "type")
+    private String regionType;
```
Comment (aklish) [inline webservice/src/main/java/com/yahoo/covid19/models/Place.java]: Doesn't this need to be a list of types?  See my comment about JsonType in Elide.

**Review by aklish — CHANGES_REQUESTED** [review]

File: db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java [hunk 7]
```diff
@@ -164,13 +159,10 @@ public Object getId() {
 
     @Override
     public String getTableName() {
-        return type == null ? "" : type.name();
+        return TABLE_NAME;
     }
 
     @Override
     public void getForiegnKeyFields(Map<String, Insertable> foreignKeyMap) {
-        if (type.equals(JoinTableNames.Supername)) {
```
Comment (aklish) [inline db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java:221]: Can we remove this entire function from INsertable?

Comment (Chandrasekar-Rajasekar) [reply db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java:221]: healthRecord uses this function to populate lat, lon, wikiId

**Review by aklish — APPROVED** [review]

**Pull request merged**